spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/factory/SpringCloudCircuitBreakerFilterFactory.java

```diff
@@ -85,7 +92,19 @@ public GatewayFilter apply(Config config) {
 			@Override
 			public Mono<Void> filter(ServerWebExchange exchange,
 					GatewayFilterChain chain) {
-				return cb.run(chain.filter(exchange), t -> {
+				return cb.run(chain.filter(exchange).doOnSuccess(v -> {
+					Set<HttpStatus> statuses = config.getStatusCodes().stream()
```

Comment:
I think this expression gathering status could be moved to line 90 below `cb = ` so that it isn't run every request, only when the filter is created with config